src/command_buffer.rs

```diff
@@ -0,0 +1,221 @@
+        storage: NonNull<u8>,
+    ) -> (NonNull<u8>, Layout) {
+        let layout = Layout::from_size_align(min_size.next_power_of_two().max(64), align).unwrap();
+        let new_storage = NonNull::new_unchecked(alloc(layout));
+        ptr::copy_nonoverlapping(storage.as_ptr(), new_storage.as_ptr(), cursor);
+        (new_storage, layout)
+    }
+
+    unsafe fn add_inner(&mut self, ptr: *mut u8, ty: TypeInfo) {
+        let offset = align(self.cursor, ty.layout().align());
+        let end = offset + ty.layout().size();
+
+        if end > self.layout.size() || ty.layout().align() > self.layout.align() {
+            let new_align = self.layout.align().max(ty.layout().align());
+            let (new_storage, new_layout) = Self::grow(end, self.cursor, new_align, self.storage);
+            if self.layout.size() != 0 {
+                dealloc(self.storage.as_ptr(), self.layout);
+            }
+            self.storage = new_storage;
+            self.layout = new_layout;
+        }
+
+        let addr = self.storage.as_ptr().add(offset);
+        ptr::copy_nonoverlapping(ptr, addr, ty.layout().size());
+        self.info.push(InfoBuffer {
+            ty_info: ty,
+            size: offset,
+        });
+        self.cursor = end;
+    }
+
+    /// Buffer 'reserved_entity' with 'bundle'
+    ///
+    /// Can also be used with 'EntityBuilder'
+    pub fn spawn_at(&mut self, ent: Entity, bundle: impl DynamicBundle) {
+        let len = bundle.type_info().len();
+        unsafe {
+            bundle.put(|ptr, ty| self.add_inner(ptr, ty));
+        }
+        let beg = self.get_beg();
+        let end = beg + len;
+        self.ent.push(EntityIndex {
+            entity: ent,
+            beg,
+            end,
+        });
+    }
+
+    fn get_beg(&self) -> usize {
+        match self.ent.len() {
+            0 => 0,
+            _ => self.ent[self.ent.len() - 1].end,
+        }
+    }
+
+    /// Spawn every `entity` recorded with their components
+    ///
+    /// Useful for recording and spawning entities at some point in the future
+    ///
+    /// # Example
+    /// ```
+    /// # use hecs::*;
+    /// let mut world = World::new();
+    /// let a = world.reserve_entity();
+    /// let mut recorder = CommandBuffer::new();
+    /// recorder.spawn_at(a, (false,0.0));
+    /// recorder.run_on(&mut world);
+    /// assert!(world.contains(a));
+    /// ```
+    pub fn run_on(&mut self, world: &mut World) {
+        let mut mark: usize = 0;
+        self.sort_buffered();
+        (0..self.ent.len()).for_each(|_| {
+            let (ent, comps) = self.build(mark);
+            world.spawn_at(ent, comps);
+            mark += 1;
+        });
+        self.clear();
+    }
+
+    pub(crate) fn sort_buffered(&mut self) {
+        for index in self.ent.iter() {
+            self.info[index.beg..index.end].sort_unstable_by_key(|z| z.ty_info);
+        }
+    }
+
+    pub(crate) fn build(&mut self, mark: usize) -> (Entity, ReadyBuffer<'_>) {
```

Comment:
This isn't needed from outside the module.
```suggestion
    fn build(&mut self, mark: usize) -> (Entity, ReadyBuffer<'_>) {
```